client/my-sites/media-library/list-item.jsx

```diff
@@ -98,7 +98,7 @@ export default class extends React.Component {
 	};
 
 	render() {
-		var classes, props, style, title;
+		let classes, props, style, title, selectedNumber;
```

Comment:
@jdlrobson this is unfortunately pretty normal.  We are working on build speed, but in the past it has taken a back seat to other priorities.

I'd suggest the following workarounds:

- Linting: run something like `./node_modules/.bin/eslint-eslines $(git diff --name-only $(git merge-base origin/master HEAD) | grep -E '\.jsx?$')` which is both faster and should have the exact same results as our CI lint checks (I've created a local script that does this).
- `npm start`: not much way around this, maybe make a shell alias that configures things as you need so that you can rely more on incremental rebuilds.
- `npm test`: only run the tests that you think may be affected by your changes (`./node_modules/.bin/jest -c=test/client/jest.config.json folder1/ folder2/test/file.js`) and wait for CI to do the rest.

Reply:
@nylen would you mind if I captured this in an issue or does one already exist that I can subscribe to. The problem I have is that I usually have an hour any given day to work with the code submit patches and I'm seeing most of that time running npm commands. 

I should state that I'm very interested in all this as I'm trying to introduce many of the technologies you are using into the Wikipedia stack.

Reply:
I think we would love to see issues that say more about some of the pain points.  After all, they are the same pain points for us.  I see a few things discussed here that are likely to have totally separate fixes:

- Our eslint situation.  It should be easier to figure out what is going on there.
- Build speed.  https://github.com/Automattic/wp-calypso/pull/7636 might help, right now you can use that in `master` if you run with `ENABLE_FEATURES=webpack/persistent-caching npm start`.  We should revisit that and make it the default.
- Tests.  I'm not really sure what to do here, running the full test suite should probably be expected to take a while.
- Better `PORT` handling, if there is anything specific to do here?

Reply:
There are some interesting issues listed in "Framework Improvements" project board under "Build": https://github.com/Automattic/wp-calypso/projects/7

Reply:
Thanks @nylen and @simison. I've just made the eslint fix and I'm just waiting for webpack to build so I can test my fix worked as planned.

That's interesting to see and reassuring (I think!?) that's it not just me, but I could see that as being a big barrier to a first patch in this repo. The eslint one is pretty guidable once pointed out, but the build process is very discouraging (if I wasn't a good multitasker I probably would have struggled with doing this eslint fix :)).

The PORT issue is my problem (I need 8142 for another Wikipedia related project so it defaults to that), but if 3000 is the only supported PORT in development, this could be detected and flagged early on. Not sure if there's a bug there?